hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutFromLoadAllOperation.java

```diff
@@ -70,8 +71,12 @@ public void run() throws Exception {
             Object previousValue = recordStore.putFromLoad(key, value);
 
             callAfterPutInterceptors(value);
+            // Null-pointer here, in case of null key loaded by MapLoader
+            checkNotNull(key, "Key loaded by a MapLoader cannot be null.");
```

Comment:
actually, the intended behaviour was not throw exception in case of null key or value, silently continue processing and print a warning message. This operation can be a continuation of that intended behaviour.  You can see this here: [isKeyAndValueLoadable](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java#L949)

Seems until 4.0 we can preserve it.